src/sentry/sentry_metrics/querying/visitors/query_expression.py

```diff
@@ -437,3 +460,79 @@ def _visit_float(self, float_number: float) -> QueryExpression:
 
     def _is_numeric_scalar(self, value: QueryExpression) -> bool:
         return isinstance(value, int) or isinstance(value, float)
+
+
+class MapperVisitor(QueryExpressionVisitor):
+    """
+    Visitor that recursively transforms the QueryExpression components to modulate certain attributes to be queried
+    by API that need to be translated for Snuba to be able to query the data.
+    """
+
+    def __init__(self, projects: Sequence[Project], mapper_config: MapperConfig):
+        self.projects = projects
+        self.mapper_config = mapper_config
+        self.mappers: list[Mapper] = []
+
+    def get_or_create_mapper(
+        self, from_key: str | None = None, to_key: str | None = None
+    ) -> Mapper | None:
+        # retrieve the mapper type that is applicable for the given key
+        mapper_class = self.mapper_config.get(from_key=from_key, to_key=to_key)
+        # check if a mapper of the type already exists
+        if mapper_class:
+            for mapper in self.mappers:
+                if mapper_class == type(mapper):
+                    # if a mapper already exists, return the existing mapper
+                    return mapper
+            else:
+                # if no mapper exists yet, instantiate the object and append it to the mappers list
+                mapper_instance = mapper_class()
+                self.mappers.append(mapper_instance)
+                return mapper_instance
+        else:
+            # if no mapper is configured for the key, return None
+            return None
```

Comment:
Since this function is defined twice, can we maybe lift it out in the `mapper` file and call it? You can just supply `mapping_config` and `mappers` as params and you should be good. Wdyt?

Reply:
Changed it as suggested